### PR TITLE
[MMI] 2510 move mmi components to mmi folder

### PR DESCRIPTION
**Pull request by zone-live**
Adds all MMI components and their respective EN locales.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -284,6 +284,7 @@
     "debounce-stream": "^2.0.0",
     "deep-freeze-strict": "1.1.1",
     "end-of-stream": "^1.4.4",
+    "enzyme": "^3.11.0",
```
Comment (brad-decker) [inline package.json:287]: You will need to rewrite all tests using React testing library, as enzyme was recently removed and we do not want to backtrack on that effort.

File: shared/modules/hash.utils.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable no-undef */
+export async function sha256(str) {
+  const buf = await crypto.subtle.digest(
+    'SHA-256',
+    new TextEncoder('utf-8').encode(str),
+  );
+
+  return Array.prototype.map
+    .call(new Uint8Array(buf), (x) => `00${x.toString(16)}`.slice(-2))
+    .join('');
```
Comment (brad-decker) [inline shared/modules/hash.utils.js:10]: All new files added to the shared folder should be typescript

File: ui/selectors/selectors.js [hunk 3]
```diff
@@ -623,6 +628,25 @@ export function getSubjectMetadata(state) {
...
+      ? getCustodyAccountSupportedChains(state, selectedIdentity.address)
+      : null;
+
+  return supportedChains?.supportedChains
+    ? supportedChains.supportedChains.includes(
+        Number(provider.chainId).toString(),
+      )
+    : true;
+}
+///: END:ONLY_INCLUDE_IN
```
Comment (brad-decker) [inline ui/selectors/selectors.js:648]: I can see why this is here and not in the duck, but can we extract this to selectors/institutional?

File: ui/store/actionConstants.ts [hunk 4]
```diff
@@ -28,6 +28,9 @@ export const DISPLAY_WARNING = 'DISPLAY_WARNING';
 export const HIDE_WARNING = 'HIDE_WARNING';
 export const CAPTURE_SINGLE_EXCEPTION = 'CAPTURE_SINGLE_EXCEPTION';
 // accounts screen
+///: BEGIN:ONLY_INCLUDE_IN(mmi)
+export const SHOW_ACCOUNT_DETAIL = 'SHOW_ACCOUNT_DETAIL';
+///: END:ONLY_INCLUDE_IN
```
Comment (brad-decker) [inline ui/store/actionConstants.ts:33]: Why was this added back for institutional?

File: ui/store/actions.ts [hunk 5]
```diff
@@ -1894,6 +1897,75 @@ export function setSelectedAddress(
   };
 }
 
+///: BEGIN:ONLY_INCLUDE_IN(mmi)
+export function showAccountDetail(
+  address: string,
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return async (dispatch, getState) => {
+    dispatch(showLoadingIndication());
```
Comment (brad-decker) [inline ui/store/actions.ts:1905]: This was recently removed, why was it added back?

File: ui/store/actions.ts [hunk 6]
```diff
@@ -4599,3 +4671,18 @@ export async function disableDesktop() {
...
+    log,
+    showLoadingIndication,
+    submitRequestToBackground,
+    displayWarning,
+    hideLoadingIndication,
+    forceUpdateMetamaskState,
+    showModal,
+
+    callBackgroundMethod,
+  });
```
Comment (brad-decker) [inline ui/store/actions.ts:4687]: can you describe what this is doing? Looks like you are passing access to our background connection to an external library which I doubt we are comfortable with.

File: ui/components/app/mmi/interactive-replacement-token-notification/interactive-replacement-token-notification.component.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { sha256 } from '../../../../../shared/modules/hash.utils';
+
+export default class InteractiveReplacementTokenNotification extends PureComponent {
```
Comment (brad-decker) [inline ui/components/app/mmi/interactive-replacement-token-notification/interactive-replacement-token-notification.component.js:5]: Please refactor all new components to be functional components, adding more class based components is increasing our technical debt.

File: ui/components/app/mmi/interactive-replacement-token-notification/interactive-replacement-token-notification.container.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { compose } from 'redux';
+import { getMMIActions } from '../../../../store/actions';
+import { getCurrentKeyring } from '../../../../selectors';
+import InteractiveReplacementTokenNotification from './interactive-replacement-token-notification.component';
+
+const mapStateToProps = (state) => {
```
Comment (brad-decker) [inline ui/components/app/mmi/interactive-replacement-token-notification/interactive-replacement-token-notification.container.js:8]: We no longer use the .component / .container format, we use functional components that make use of useSelector / useDispatch. Please refactor before this can merge.

File: ui/components/app/mmi/interactive-replacement-token-notification/interactive-replacement-token-notification.component.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    this.setState({
+      showNotification:
+        showNotification &&
+        refreshTokenAccount === interactiveReplacementToken.oldRefreshToken,
+    });
+  }
+
+  render() {
+    return this.state.showNotification ? (
+      <div className="interactive-replacement-token-notification">
```
Comment (brad-decker) [inline ui/components/app/mmi/interactive-replacement-token-notification/interactive-replacement-token-notification.component.js:95]: Please use our design system component "Box" to achieve a lot of styles. you may still need css but it should be done only when the props for Box don't exist for what you are needing.

File: ui/components/app/mmi/modals/compliance-modal/compliance-modal.component.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+      <Modal
+        onClose={this.handleClose}
+        onSubmit={this.handleSubmit}
+        submitText={t('openCodefiCompliance')}
+        submitType="primary"
+        containerClass="compliance-modal-container"
+      >
+        <div className="compliance-modal">
+          <div className="compliance-modal__header">
+            <div className="compliance-modal__header-logo">
```
Comment (brad-decker) [inline ui/components/app/mmi/modals/compliance-modal/compliance-modal.component.js:37]: These should be boxes.

File: ui/components/app/mmi/modals/compliance-modal/compliance-modal.component.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+            </div>
+            <div
+              className="compliance-modal__header-close"
+              onClick={this.handleClose}
+              data-testid="compliance-modal-close"
+            />
+          </div>
+
+          <br />
+          <p>{t('complianceBlurb0')}</p>
```
Comment (brad-decker) [inline ui/components/app/mmi/modals/compliance-modal/compliance-modal.component.js:53]: Please use the "Text" component from 'component-library'

**Review by brad-decker — CHANGES_REQUESTED** [review]
This Pull request is going to need a lot of work to get it into a place where we can merge it. I've done a first pass here.

1. Instead of components/mmi/app and components/mmi/ui I suggest you create a single components/institutional folder that contains all of your components. the UI/app separation is not something we are committed to extend at this time. Further more even if we were to separate them the way we do currently, none of the components in UI are really UI components except maybe drag-and-drop... icons will have to be refactored to use our iconography/icon system[
2. @georgewrmarshall is our design system lead he will give you some guidance on the icons and general design system usage.
3. We can't merge react components that use our old patterns, any other PR would be asked for refactors to our current patterns including functional components, not using .component/.container separation, utilizing our design system components, etc. I realize this is essentially a long running feature branch and the code was written prior to a lot of these conventions but alas we are here now at the junction point (and I am excited!) but we need to make sure we don't over burden the extension with debt.
4. I'd like to see the duck you introduced to be consolidated to ducks/institutional instead of the mmi folder intermediary. I don't think including the abbreviation of MMI is helpful to anyone and creates further folder depth and confusion. institutional is fine. I would also encourage refactoring the institutional duck to use reduxjs/toolkit, which you can see in usage in the send and alerts duck. 
5. pages 'mmi' folder should just be institutional instead of mmi.
6. We deprecated enzyme, please remove and refactor all tests to use React testing library.
7. Probably more after I do further passes.

Please also break this PR into chunks but leave this one open in draft so that we can track against these change requests

**Review by georgewrmarshall — COMMENTED** [review]
Hello! We have done a lot of work to ensure all of our components are dark mode compatible as well as have consistent colors across extension and mobile platforms using the https://github.com/MetaMask/design-tokens repo. Would it be possible to update any static hex values and non design token CSS variables to use the design token CSS variables?

**Comment by zone-live** [comment]
Thank you for the valuable feedback 💪🏼 we will be doing this refactor to our components & tests in the following weeks, and come back with individual PRs for each, to be easier to review as well.
